lib/api/apiUtils/object/tagging.js

```diff
@@ -0,0 +1,143 @@
+import { errors } from 'arsenal';
+import { parseString } from 'xml2js';
+
+const tagRegex = new RegExp(/[^a-zA-Z0-9 +-=._:/]/g);
+
+/*
+    Format of xml request:
+
+    <Tagging>
+      <TagSet>
+         <Tag>
+           <Key>Tag Name</Key>
+           <Value>Tag Value</Value>
+         </Tag>
+      </TagSet>
+    </Tagging>
+*/
+
+
+export const _validator = {
+    validateTagStructure: tag => tag
+        && Object.keys(tag).length === 2
+        && tag.Key && tag.Value
+        && tag.Key.length === 1 && tag.Value.length === 1
+        && tag.Key[0] && tag.Value[0]
+        && typeof tag.Key[0] === 'string' && typeof tag.Value[0] === 'string',
+
+    validateXMLStructure: result =>
+        result && Object.keys(result).length === 1 &&
+        result.Tagging &&
+        result.Tagging.TagSet &&
+        result.Tagging.TagSet.length === 1 &&
+        (
+          result.Tagging.TagSet[0] === '' ||
+          result.Tagging.TagSet[0] &&
+          Object.keys(result.Tagging.TagSet[0]).length === 1 &&
+          result.Tagging.TagSet[0].Tag &&
+          Array.isArray(result.Tagging.TagSet[0].Tag)
+        ),
+
+    validateKeyValue: (key, value) => {
+        if (key.length > 128 || key.match(tagRegex)) {
+            return errors.InvalidTag.customizeDescription('The TagKey you ' +
+              'have provided is invalid');
+        }
+        if (value.length > 256 || value.match(tagRegex)) {
+            return errors.InvalidTag.customizeDescription('The TagValue you ' +
+              'have provided is invalid');
+        }
+        return true;
+    },
+};
+/** _validateTags - Validate tags, returning an error if tags are invalid
+* @param {object[]} tags - tags parsed from xml to be validated
+* @param {string[]} tags[].Key - Name of the tag
+* @param {string[]} tags[].Value - Value of the tag
+* @return {(Error|object)} - return tags object on success; error on failure
```

Comment:
actually returning an array of objects.  please update and state what the format of the objects in the array is.